Patterns/DependencyInjection.md

```diff
@@ -121,3 +126,21 @@ If every implementation is the same, use the following instead:
 services.AddTransient(typeof(IRepository<>), typeof(Repository<>));
 ```
 
+### Registering `internal` services
+
+When building class libraries it is a good practice to shield library specific details away from other projects by marking some classes as `internal`. A service in `Contoso.Services` just needs `IRepository<SomeEntity>` from `Contoso.Repositories` and should not be able to touch `SomeEntityRepositoryImplementation`. But when registering that repository in the DI container `Contoso.API` a developer can no longer do `services.AddScoped<IRepository<SomeEntity>, SomeEntityRepositoryImplementation>()` as `Contoso.API` cannot access the internals of `Contoso.Repositories`. To solve this, every class library should have a `DependencyConfiguration.cs` file at the root of its project, containing something like this:
```

Comment:
The part "..to **shield** library specific details **away**..." could be better formulated by changing it to:
"..to use [access modifiers](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers) on library specific details to control their visibility to other projects.".

>> I'd use the terms access modifier here to indicate whether these classes can be used from other assemblies, instead of using 'shield away'.